lib/mri/mkmf.rb

```diff
@@ -552,7 +552,7 @@ def try_link0(src, opt="", *opts, &b) # :nodoc:
       end
     else
       try_do(src, cmd, *opts, &b)
-    end and File.executable?(CONFTEST+$EXEEXT)
+    end and File.exist?(CONFTEST+$EXEEXT) # TruffleRuby: TRY_LINK generates a non-executable file containing bitcode
```

Comment:
@bjfish This seems a proper fix for try_link.
Alternatively we could try to not pass `-emit-llvm` for TRY_LINK and compile a real executable as this is mostly used to check if library macros/methods are defined.